source/MaterialXRender/OpenGL/GLTextureHandler.cpp

```diff
@@ -0,0 +1,149 @@
+#include <MaterialXCore/Types.h>
+#include <MaterialXRender/OpenGL/GLTextureHandler.h>
+#include <MaterialXRender/OpenGL/GlslProgram.h>
+#include <MaterialXRender/External/GLew/glew.h>
+
+namespace MaterialX
+{
+bool GLTextureHandler::createColorImage(const MaterialX::Color4& color,
+                                        ImageDesc& imageDesc)
+{
+    ParentClass::createColorImage(color, imageDesc);
+    if ((imageDesc.width * imageDesc.height > 0) && imageDesc.resourceBuffer)
+    {
+        glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
+        glGenTextures(1, &imageDesc.resourceId);
+        glActiveTexture(GL_TEXTURE0 + imageDesc.resourceId);
+        glBindTexture(GL_TEXTURE_2D, imageDesc.resourceId);
+        glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA32F, imageDesc.width, imageDesc.height, 0, GL_RGBA, GL_FLOAT, imageDesc.resourceBuffer);
+        glGenerateMipmap(GL_TEXTURE_2D);
+        glBindTexture(GL_TEXTURE_2D, 0);
+
+        return true;
+    }
+    return false;
+}
+
+bool GLTextureHandler::getImage(std::string& fileName,
```

Comment:
"Getting" an image is detached from binding an image. This allows for the image to be created, the client to perform binding to a uniform on it's own shader program (GLSLprogram, nano-gui GLShader), and then the texture is bound to that location.